core/src/main/java/org/elasticsearch/env/Environment.java

```diff
@@ -126,6 +128,11 @@ public Environment(Settings settings) {
             dataFiles = new Path[]{homeFile.resolve("data")};
             dataWithClusterFiles = new Path[]{homeFile.resolve("data").resolve(ClusterName.clusterNameFromSettings(settings).value())};
         }
+        if (settings.get("path.shared_data") != null) {
+            sharedDataFile = PathUtils.get(cleanPath(settings.get("path.shared_data")));
+        } else {
+            sharedDataFile = homeFile.resolve("data");
```

Comment:
Then maybe it should be null when not set, otherwise we request an unnecessary permission to the security manager, while we should be requesting as few permissions as possible?